modules/renter/hostdb/online.go

```diff
@@ -0,0 +1,39 @@
+package hostdb
+
+import (
+	"time"
+)
+
+// online.go performs regular checks against the gateway to see if there is
+// internet connectivity. The check is performed by looking for at least one
+// non-local peer in the peer list. If there is no internet connectivity, scans
+// must be stopped lest we penalize otherwise online hosts.
+
+func (hdb *HostDB) threadedOnlineCheck() {
```

Comment:
The depedencies are already there, the hostdb already needs a consensus set which needs a gateway. While they are now more visible, it doesn't change the dependency graph too much.

I don't think it's likely to cause problems, because the gateway inherently aggressively looks for nonlocal outbound peers. Would be a very different network if somehow it didn't need to be doing that.